server/src/test/java/org/elasticsearch/cluster/metadata/AliasMetaDataTests.java

```diff
@@ -52,4 +53,49 @@ public void testSerialization() throws IOException {
 
         assertThat(after, equalTo(before));
     }
+
+    @Override
+    protected AliasMetaData createTestInstance() {
+        return createTestItem();
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return p -> p.equals("") // do not add elements at the top-level as any element at this level is parsed as a new alias
+                || p.contains(".filter"); // do not insert random data into AliasMetaData#filter
+    }
+
+    @Override
+    protected AliasMetaData doParseInstance(XContentParser parser) throws IOException {
+        if (parser.nextToken() == XContentParser.Token.START_OBJECT) {
+            parser.nextToken();
+        }
+        assertEquals(XContentParser.Token.FIELD_NAME, parser.currentToken());
+        AliasMetaData aliasMetaData = AliasMetaData.Builder.fromXContent(parser);
+        assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+        return aliasMetaData;
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
```

Comment:
could you add a comment that clarifies this is needed as this class is used by the high-level REST client. I am quite worried that this class is internal (like many others though) yet its parsing may be modified without taking the client into account.

Reply:
@javanna I can add a comment later today or the commit msg is enough?
Sorry for getting back to you that late.

Reply:
no worries at all, it was not that important. and I was not willing to wait for another test run after adding the comment addition so I merged :)